web/src/OverviewResourceSidebar.tsx

```diff
@@ -28,7 +29,11 @@ export default function OverviewResourceSidebar(
   let logAlertIndex = useLogAlertIndex()
   let pathBuilder = usePathBuilder()
   let resources = props.view.uiResources || []
-  let items = resources.map((res) => new SidebarItem(res, logAlertIndex))
+  let buttons = buttonsByComponent(props.view.uiButtons)
```

Comment:
do you think it's worth memoizing the button map?

Reply:
Yeah, that's probably a good idea.